src/Service/Nph/NphOrderService.php

```diff
@@ -893,4 +893,58 @@ public function getVisitTypes(): array
     {
         return $this->moduleObj->getVisitTypes();
     }
+
+    public function validateGenerateOrdersData(array $formData): array
+    {
+        $formErrors = [];
+        $hasSample = false;
+        foreach (array_keys($this->getTimePoints()) as $timePoint) {
+            if (!empty($formData[$timePoint])) {
+                $hasSample = true;
+                break;
+            }
+        }
+        if (empty($formData['stoolKit'])) {
+            if ($hasSample === false) {
+                $formErrors[] = [
+                    'field' => 'checkAll',
+                    'message' => 'Please select or enter at least one sample'
+                ];
+                return $formErrors;
+            }
+        }
+        if (!empty($formData['stoolKit'])) {
```

Comment:
It looks like this could be simplified into an `if ... else` 

```suggestion
        if (empty($formData['stoolKit'])) {
            if ($hasSample === false) {
                $formErrors[] = [
                    'field' => 'checkAll',
                    'message' => 'Please select or enter at least one sample'
                ];
                return $formErrors;
            }
        }
       else {
```